software/base/src/main/java/brooklyn/entity/software/MachineLifecycleEffectorTasks.java

```diff
@@ -561,7 +560,7 @@ public void stop(ConfigBag parameters) {
 
         boolean isEntityStopped = entity().getAttribute(SoftwareProcess.SERVICE_STATE_ACTUAL)==Lifecycle.STOPPED;
 
-        DynamicTasks.queue("pre-stop", new Callable<String>() { public String call() {
+        DynamicTasks.queue("pre-stop of " + entity() + ", submitted by " + Tasks.current(), new Callable<String>() { public String call() {
```

Comment:
looking at that comment -- the task name is `pre-stop` but it shows `pre-stop of <entity>, submitted by Task[<ibid>]` (recursing).  where did the extra log info `of ..., submitted by ...` come from in that case?  because if i understand, when you *add* extra data to the task name, and it shows *less* information, i.e. just the name `pre-stop of <entity>, submitted by <other-task>` and not that extra log info.

my gut is saying this isn't the best way to address poor logging, as it would suggest we should apply similar enriched data to most tasks.